source/js/l10n.js

```diff
@@ -0,0 +1,38 @@
+import "core-js/stable";
+import "regenerator-runtime/runtime";
+import "intl-pluralrules";
+import { FluentBundle, FluentResource } from "@fluent/bundle/compat";
+import { negotiateLanguages } from "@fluent/langneg";
+import { getCurrentLanguage } from "./components/petition/locales";
+
+import deMessages from "../../network-api/locale/de/messages.ftl";
+import esMessages from "../../network-api/locale/es/messages.ftl";
+import frMessages from "../../network-api/locale/fr/messages.ftl";
+import plMessages from "../../network-api/locale/pl/messages.ftl";
+import ptMessages from "../../network-api/locale/pt_BR/messages.ftl";
+import sourceMessages from "../../network-api/locale/templates/messages.ftl";
+
+const LOCALES = ["de", "en", "es", "fr", "pl", "pt"];
+
+const MESSAGES_ALL = {
+  de: deMessages,
+  en: sourceMessages,
+  es: esMessages,
+  fr: frMessages,
+  pl: plMessages,
+  pt: ptMessages
+};
+
+export function* generateBundles() {
```

Comment:
what is the benefit of a generator here?

Reply:
This is a Fluent-ism, too: https://github.com/projectfluent/fluent.js/wiki/LocalizationProvider

It's not strictly required, and a simple array would work here as well. The benefit of using the generator is that it yields bundles lazily, so if all translations are found in the first bundle (the user's most preferred language), you won't pay the cost of parsing the translations in other languages.

Reply:
Thanks, replaced this with a array since it allows us to drop two dependencies and we’re not really doing language negotiation anyway, we just have to follow the Django locale. We’re only generating bundles for "en" + the current page locale, so there wasn’t much of a performance gain in the first place.